File: ColorImage.cpp
#include "opencv2/core/core.hpp"
#include "opencv2/highgui/highgui.hpp"
#include "opencv2/imgproc/imgproc.hpp"
#include "iostream"

 // Author Rishab Shah

using namespace cv;
using namespace std;
 
int main(int argc, char** argv )
{
    Mat src;
    src = imread(argv[1], CV_LOAD_IMAGE_COLOR);           //load the image
    namedWindow( "Original image", CV_WINDOW_AUTOSIZE );  //open a window
    imshow( "Original image", src);                       //show images

	vector<Mat> input_planes(3);
	split(src,input_planes);                              //split into three channels
	Mat channel1_display, channel2_display, channel3_display;
        imshow("Red",   input_planes[2]);
        imshow("Green",   input_planes[1]);
        imshow("Blue",   input_planes[0]);


	Mat ycrcb_image;
	cvtColor(src, ycrcb_image, CV_BGR2YCrCb);            //convert BGR into YCrCb
	split(ycrcb_image,input_planes);                     //split into three channels
        imshow("Y",   input_planes[0]);
        imshow("Cb",   input_planes[1]);
        imshow("Cr",   input_planes[2]);


	Mat hsv_image;
	cvtColor(src, hsv_image, CV_BGR2HSV);               //convert BGR into HSV
	vector<Mat> hsv_planes(3);                          //split into three channels
	split(hsv_image,hsv_planes);
        imshow("Hue",   hsv_planes[0]);
        imshow("Saturation",   hsv_planes[1]);
        imshow("Value",   hsv_planes[2]);


	
    waitKey(0);                                       
    return 0;
} 

File: Exercise2/ColorImage.py
# -*- coding: utf-8 -*-
"""
Spyder Editor

This is a temporary script file.
"""

import cv2

lena = cv2.imread('Test_images/Lenna.png') #load Lenna image

[b,g,r] = cv2.split(lena) #split into color channels

cv2.imshow('Blue',b)
cv2.imwrite('Blue.png',b)
cv2.imshow('Red',r) 
cv2.imwrite('Red.png',r)
cv2.imshow('Green',g)
cv2.imwrite('Green.png',g)

print(lena[20,25])

hsv_lena = cv2.cvtColor(lena, cv2.COLOR_BGR2HSV) #convert to HSV

[h,s,v] = cv2.split(hsv_lena) #split into HSV channels

cv2.imshow('Hue',h)
cv2.imwrite('Hue.png',h)
cv2.imshow('Saturation',s)
cv2.imwrite('Saturation.png',s)
cv2.imshow('Value',v)
cv2.imwrite('Value.png',v)

print(hsv_lena[20,25])

YCC_lena = cv2.cvtColor(lena, cv2.COLOR_BGR2YCR_CB) #convert to YCrCb

[y,Cb,Cr] = cv2.split(YCC_lena) #split YCrCb channels

cv2.imshow('Y',y)
cv2.imwrite('Y.png',y)
cv2.imshow('Cb',Cb)
cv2.imwrite('Cb.png',Cb)
cv2.imshow('Cr',Cr)
cv2.imwrite('Cr.png',Cr)

print(YCC_lena[20,25])

cv2.waitKey(0)
